tests/tests.py

```diff
@@ -938,3 +951,120 @@ def test_invalidate_model_with_geometry(self):
         geom.save()
         # Raises ValueError if this doesn't work
         invalidate_obj(geom)
+
+
+class ThreadWithReturnValue(Thread):
+    def __init__(self, *args, **kwargs):
+        super(ThreadWithReturnValue, self).__init__(*args, **kwargs)
+        self._return = None
+
+    def run(self):
+        if six.PY3:
+            if self._target is not None:
+                self._return = self._target(*self._args, **self._kwargs)
+        if six.PY2:
+            if self._Thread__target is not None:
+                self._return = self._Thread__target(*self._Thread__args, **self._Thread__kwargs)
+
+    def join(self, *args, **kwargs):
+        super(ThreadWithReturnValue, self).join(*args, **kwargs)
+        return self._return
+
+
+def return_from_other_thread(target, **kwargs):
+    @wraps(target)
+    def wrapper(*args, **wrapper_kwargs):
+        try:
+            return target(*args, **wrapper_kwargs)
+        except Exception as e:
+            # make sure other thread exceptions make it out to the test thread, instead of having
+            # None returned and the test thread continuing.
+            return e
+        finally:
+            # django does not drop postgres connections opened due to new threads. results in
+            #  Postgres complaining about connected users when django tries to delete test db
+            #  See https://code.djangoproject.com/ticket/22420#comment:18
+            from django.db import connection
+            connection.close()
```

Comment:
using wrapper now in 6b214457. I also removed the finally part since it clearly wasn't needed (at least, anymore)